src/main/java/org/springframework/data/jpa/repository/query/StoredProcedureAttributes.java

```diff
@@ -49,14 +54,31 @@
 	 */
 	public StoredProcedureAttributes(String procedureName, @Nullable String outputParameterName,
 			Class<?> outputParameterType, boolean namedStoredProcedure) {
+		this(procedureName, Arrays.asList(outputParameterName), Arrays.asList(outputParameterType), namedStoredProcedure);
+	}
+
+	/**
+	 * Creates a new {@link StoredProcedureAttributes}.
+	 *
+	 * @param procedureName must not be {@literal null}
+	 * @param outputParameterNames may be empty, but not null
+	 * @param outputParameterTypes must not be empty, and cannot be a single element of null
```

Comment:
Why are we only concerned about the first element of the return types? What would be the meaning of a null element anywhere else in the list?

Reply:
Because on line 54 that is how the original functionality worked. So this is to support anyone that was using this functionality originally so it doesn't break the pre-existing contract